File: DSA/Vyom-Singhal/Day5/Day5-q1.cpp
#include<iostream>
using namespace std;

class Node {
    public :
        int data;
        Node* next;
    void createList(Node* head, int n) {
        Node *temp = new Node();
        int data;
        cout << "Enter data for element 1 : ";
        cin >> data;
        head->data = data;
        head->next = NULL;
        temp = head;
        for(int i = 1; i < n; i++) {
            Node *newNode = new Node();
            cout << "Enter data for element " << i+1 << " : ";
            cin >> data;
            newNode->data = data;
            newNode->next = NULL;
            temp->next = newNode;
            temp = newNode;
        }
    }
    void printList(Node* head) {
        while(head != NULL) {
            cout << head->data << " ";
            head = head->next;
        }
    }
    void reverseList(Node*& head) {
        Node* current = head;
        Node* prev = head;
        Node* next = current->next;
        prev->next = NULL;
        while(next != NULL) {
            head = next;
            current = head;
            next = current->next;
            current->next = prev;
            prev = current;
        }
    }
};

int main() {
    Node* head = new Node();
    int n;
    cout << "Enter number of elements : ";
    cin >> n;
    head->createList(head,n);
    head->reverseList(head);
    head->printList(head);
}

File: DSA/Aanidhay-Aggarwal/Day-5/Day5_q1.java
import java.util.Scanner;

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    Node head;

    // Function to add a new node to the end of the list
    void addNode(int data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
        } else {
            Node temp = head;
            while (temp.next!= null) {
                temp = temp.next;
            }
            temp.next = newNode;
        }
    }

    void printReverse() {
        printReverse(head);
    }

    void printReverse(Node node) {
        if (node == null) {
            return;
        }
        printReverse(node.next);
        System.out.print(node.data + " ");
    }
}

public class Main {
    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();

        System.out.println("Enter the elements:");
        for (int i = 0; i < n; i++) {
            int data = scanner.nextInt();
            list.addNode(data);
        }

        System.out.print("Linked list: ");
        Node temp = list.head;
        while (temp!= null) {
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.println("null");

        System.out.print("Output: ");
        list.printReverse();
        System.out.println();
    }
}
